### PR TITLE
Make runnable on Windows OS

### PR DESCRIPTION
**Pull request by odunayo12**
It was not smooth replicating the application on `windows` machine (See errors generated [here](https://github.com/odunayo12/dwnld-n-save-UTube/blob/master/test.ipynb)). So in trying to find a solution I debugged the code. I hope it functions optimally on `mac os` as well. The following were the issues I found and resolved.
- [x]  Add save option. Now the user can enter the path, where to save the downloaded file via input. But defaults to the working directory.
- [x]  Correct `FileNotFoundError` on `Windows os` via  `subprocess.call()`.
- [x] Reduce time taken to collect download data, especially if download `URL`  is from a playlist.

### CODE REVIEW

File: run.py [hunk 1]
```diff
@@ -9,6 +13,32 @@ def run():
     video_info = youtube_dl.YoutubeDL().extract_info(
         url=video_url, download=False
     )
+
+    vedio_title = video_info['title']
```
Comment (imdadahad) [inline run.py]: Small typo here, should be `video_title`

File: run.py [hunk 2]
```diff
@@ -9,6 +13,32 @@ def run():
...
+    filename = f"{vedio_title}.mp3"
+
+    # Ask the user for the path they want to save the file after it has been downloaded
+    path_to_save = input(
+        r"Enter where the file is to be saved or leave empty if you want it to be saved in the current directory: ")
+    if path_to_save == "":
+        where_to_save = os.getcwd()
+    else:
+        where_to_save = f"{path_to_save}"
```
Comment (imdadahad) [inline run.py:27]: Can you declare `where_to_save` above line 21, this way you can do a `!=` check and can get rid of the `else` check

File: run.py [hunk 3]
```diff
@@ -26,6 +56,15 @@ def run():
     # Open the file once it has been downloaded
     subprocess.call(["open", filename])
 
+    # Open the file once it has been downloaded (on Mac os)
+    try:
+        subprocess.call(["open", filename])
+    except FileNotFoundError:
+        # this will take care of the windows part
+        subprocess.call(('cmd', '/C', 'start', '', filename))
+
```
Comment (imdadahad) [inline run.py:71]: I think it's better if you explicitly check whether the user is using Windows, you can do this with the OS library: https://stackoverflow.com/questions/1325581/how-do-i-check-if-im-running-on-windows-in-python

**Review by imdadahad — COMMENTED** [review]
Great stuff, always a big fan of pull requests. Just a few comments on things I think you can change around before we merge this in 👍

File: run.py (hunk 3 above)
Comment (odunayo12) [reply run.py:71]: That was helpful.  It was cool to use `os.path()` on windows and `subprocess.call()` on `mac` os.

**Comment by odunayo12** [comment]
Done as requested. Thanks for the opportunity!

File: run.py [hunk 4]
```diff
@@ -14,19 +13,20 @@ def run():
...
-    vedio_title = video_info['title']
-    filename = f"{vedio_title}.mp3"
+    video_title = video_info['title']
+    filename = f"{video_title}.mp3"
 
     # Ask the user for the path they want to save the file after it has been downloaded
     path_to_save = input(
         r"Enter where the file is to be saved or leave empty if you want it to be saved in the current directory: ")
+
+    where_to_save = f"{path_to_save}"
```
Comment (imdadahad) [inline run.py]: Think you mean `where_to_save = os.getcwd()`?

File: run.py [hunk 5]
```diff
@@ -14,19 +13,20 @@ def run():
...
+
     if path_to_save == "":
```
Comment (imdadahad) [inline run.py]: If you add the above, here you would do `if path_to_save != ""` to check that it's not empty. If it is not, then you would have `where_to_save = f"{path_to_save}"`?

File: run.py [hunk 6]
```diff
@@ -53,18 +53,17 @@ def run():
...
+    coding_env = os.name
 
-    # Open the file once it has been downloaded (on Mac os)
-    try:
-        subprocess.call(["open", filename])
-    except FileNotFoundError:
-        # this will take care of the windows part
-        subprocess.call(('cmd', '/C', 'start', '', filename))
+    # Open the file once it has been downloaded
+    os.startfile(filename) if coding_env == 'nt' else subprocess.call(
```
Comment (imdadahad) [inline run.py:69]: Nice

**Review by imdadahad — CHANGES_REQUESTED** [review]
Hey man, looks really good - just one more thing and I think it's good to go 🙏

File: run.py (hunk 5 above)
Comment (odunayo12) [reply run.py]: I think they pretty much do the same thing. But your suggestion looks more readable and easier to follow logically. Worth adopting! 😉

File: run.py (hunk 6 above)
Comment (odunayo12) [reply run.py:69]: Thanks

**Comment by odunayo12** [comment]
Check if it serves now. Thnaks

**Comment by imdadahad** [comment]
Solid, great work and initiative @odunayo12 

Could you mention your work on a comment in the YouTube video (https://youtu.be/FsNPdhWmil4). This way others who are using Windows can see it 👍

**Pull request merged**